client_cli/pytests/src/client_cli/client_cli.py

```diff
@@ -384,6 +429,50 @@ def _execute_single(self, command):
             self.transaction_hash = extract_hash(self.stdout)
             self._attach_allure_reports()
 
+    def wait_for_transaction_commit(self, transaction_hash, timeout=1):
+        """
+        Waits for the transaction with the given hash to be committed in all configs.
+        """
+
+        def is_transaction_committed():
+            return self.is_transaction_committed(transaction_hash)
+
+        try:
+            self.wait_for(is_transaction_committed, timeout)
+            return True
+        except TimeoutError:
+            return False
+
+    def is_transaction_committed(self, transaction_hash):
+        """
+        Checks if the transaction with the given hash is committed in all configs.
+        """
+        with self.event_data_lock:
+            for config_path, data in self.event_data.items():
+                if not self._check_commit_in_output(transaction_hash, data):
+                    return False
+        return True
+
+    def _check_commit_in_output(self, transaction_hash, output):
+        """
+        Parses the output to check if the transaction with the given hash is committed.
+        """
+        decoder = JSONDecoder()
+        idx = 0
+        try:
+            while idx < len(output):
+                obj, idx_next = decoder.raw_decode(output[idx:])
+                if (
+                    obj.get("Pipeline", {}).get("entity_kind") == "Transaction"
+                    and obj.get("Pipeline", {}).get("status") == "Committed"
+                    and obj.get("Pipeline", {}).get("hash") == transaction_hash
+                ):
+                    return True
```

Comment:
I am not sure about the version of Python we are using. Can't we use [pattern matching](https://peps.python.org/pep-0636/) here?

```suggestion
                match obj:
                    case { "Pipeline": { "entity_kind": "Transaction", "status": "Committed", "hash": hash } } if hash == transaction_hash:
                        return True
```